docs/ref/contrib/admin/index.txt

```diff
@@ -1465,6 +1465,38 @@ templates used by the :class:`ModelAdmin` views:
 
     See also :ref:`saving-objects-in-the-formset`.
 
+.. warning::
+
+    All hooks that return a ``ModelAdmin`` property return the property itself
+    rather than a copy of its value. Dynamically modifying the value can lead
+    to surprising results.
+
+    Let's take :meth:`ModelAdmin.get_readonly_fields` as an example::
+
+        class PersonAdmin(admin.ModelAdmin):
+            readonly_fields = ["name"]
```

Comment:
I added these back in, as after testing found that `readonly_fields` is an empty tuple by default (and therefore immutable) - sorry about that